src/aces/query.py

```diff
@@ -137,6 +139,23 @@ def query(cfg: TaskExtractorConfig, predicates_df: pl.DataFrame) -> pl.DataFrame
         )
         to_return_cols.insert(1, "label")
 
+        if result["label"].n_unique() == 1:
+            logger.warning(
+                f"All labels in the extracted cohort are the same: '{result['label'][0]}'. "
+                "This may indicate an issue with the task logic. "
+                "Please double-check your configuration file if this is not expected."
+            )
+        else:
```

Comment:
Honestly, I don't think you need this. This can be done post-hoc, outside of ACES. I like the warning above, but this part I'm not sure you need.